frontend/src/app/Settings/Users/ManageUsers.test.tsx

```diff
@@ -0,0 +1,156 @@
+import { render, fireEvent, waitFor } from "@testing-library/react";
+import { MemoryRouter } from "react-router-dom";
+import { displayFullNameInOrder } from "../../utils";
+import ManageUsers, { SettingsUsers } from "./ManageUsers";
+
+const organization = { testingFacility: [{ id: "a1", name: "Foo Org" }] };
+const allFacilities = [
+  organization.testingFacility[0],
+  { id: "a2", name: "Bar Org" },
+];
+
+const loggedInUser = {
+  firstName: "Bob",
+  middleName: "",
+  lastName: "Bobberoo",
+  id: "b1",
+  email: "bob@bobberoo.org",
+  suffix: "",
+  roleDescription: "Admin user",
+};
+
+const users: SettingsUsers[keyof SettingsUsers][] = [
+  {
+    firstName: "John",
+    middleName: "",
+    lastName: "Doe",
+    id: "a123",
+    email: "john@doe.org",
+    organization,
+    permissions: ["READ_PATIENT_LIST"],
+    roleDescription: "user",
+  },
+  { ...loggedInUser, permissions: [], organization, roleDescription: "admin" },
+];
+
+let updateUserRole: () => Promise<any>;
+let addUserToOrg: () => Promise<any>;
+let deleteUser: (obj: any) => Promise<any>;
+let getUsers: () => void;
+
+let inputValue = (value: string) => ({ target: { value } });
+
+describe("ManageUsers", () => {
+  beforeEach(() => {
+    updateUserRole = jest.fn(() => Promise.resolve());
+    addUserToOrg = jest.fn(() => Promise.resolve());
+    deleteUser = jest.fn((obj) =>
+      Promise.resolve({ data: { setUserIsDeleted: { id: obj.variables.id } } })
+    );
+    getUsers = jest.fn();
+  });
+  it("displays the list of users and defaults to the first user", () => {
+    const { container } = render(
+      <ManageUsers
+        users={users}
+        loggedInUser={loggedInUser}
+        allFacilities={allFacilities}
+        updateUserRole={updateUserRole}
+        addUserToOrg={addUserToOrg}
+        deleteUser={deleteUser}
+        getUsers={getUsers}
+      />
+    );
+    expect(container).toMatchSnapshot();
+  });
+  it("disables logged-in user's settings", async () => {
+    const { container, findByText, getByText } = render(
+      <ManageUsers
+        users={users}
+        loggedInUser={loggedInUser}
+        allFacilities={allFacilities}
+        updateUserRole={updateUserRole}
+        addUserToOrg={addUserToOrg}
+        deleteUser={deleteUser}
+        getUsers={getUsers}
+      />
+    );
+
+    fireEvent.click(getByText(displayFullNameInOrder("Bob", "", "Bobberoo")));
+    await findByText("YOU");
+    expect(container).toMatchSnapshot();
```

Comment:
One interesting side-effect of using the utility classes is that it affects the snapshots if you change padding etc. I guess that's good?

Reply:
I think I should just remove the snapshot tests altogether and assert certain things about the DOM with react-testing-library. For example, I could assert that all the input controls are disabled (i.e., you can't change your own permissions). What do you think?